### ISSUE TITLE
just some good news

### ISSUE THREAD

**Issue by tunnckoCore** [kaerus-component/uP#4]
Really fast. For that I fork Bluebird's benchmarks to separate repo and module and announce there https://github.com/petkaantonov/bluebird/issues/372

[tunnckoCore/promise-bench](http://j.mp/1wNqmUT)

```
$ npm install promise-bench
```

[kaerus-component/uP](https://github.com/kaerus-component/uP) is second in parallel tests after bluebird.
[lvivski/davy](https://github.com/lvivski/davy) is most stable compliant after Bluebird (4-5th place)

Also module [native-or-another](https://github.com/tunnckoCore/native-or-another) (same as native-or-bluebird) checks for `micropromise`

**Comment by andelo** [comment]
oh, cool, thanks!
Maybe I should do some more optimization then ;)

**Comment by tunnckoCore** [comment]
Mm yep. But its good only on `parallel` tests, that I dont know what mean, because the code between `parallel` and `doxbee` is almost the same. Very ugly codebase at all, lol. I just dont want to make reverse engineering.

**Comment by andelo** [comment]
Perfomance depends much on how you instantiate the promise

```
p = new Promise()
```

 or

```
Promise(function(f,r,p){...});
```

where the first one should be much faster since I do 4 Function binds in the latter case, which can be done better.

**Comment by andelo** [comment]
should do the tests in browser too.

**Comment by tunnckoCore** [comment]
I think also depends on next-ticking - setTimeout/nextTick/setImmediate and etc. https://github.com/calvinmetcalf/immediate - this is awesome

**Comment by andelo** [comment]
well I have considered that -> kaerus-component/microTask
But maybe that can be done better, I'll have a look.

**Comment by andelo** [comment]
hmm... I'm going nowhere with these simple optimisations. 
Need to look more into usage patterns to gain better performance.

**Comment by tunnckoCore** [comment]
haha, its hard.
Yea, I create https://github.com/tunnckoCore/catchy pretty easy and same behaving, but cant expose compliant api, lol. Also must add tests to it ASAP, PRs welcome ;]

I'll review your PR to https://github.com/tunnckoCore/promise-bench ASAP.

**Comment by andelo** [comment]
The test facility looks fishy to me. 
Have you looked at `promises-benchmark` ?
Would be nice to have bluebird there to benchmark against.

**Comment by tunnckoCore** [comment]
You mean `promise-bench`? Yea i looked, and merge, and publish to npm.

And there have bluebird https://github.com/tunnckoCore/promise-bench/blob/master/package.json#L33

**Comment by andelo** [comment]
https://github.com/kaerus/promises-benchmark.git

**Comment by tunnckoCore** [comment]
and? I cant gotcha ... Cant understand for what we talkin about LOL

**Comment by andelo** [comment]
no worries, well I've atleast concluded that I need to rewrite micropromise to gain better performance.
It seems to be a consensus using promises through a resolver function and micropromises is not designed primarly for that use case.
